lib/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -474,6 +472,120 @@ public static function get_user_data_from_wp_global_styles( $theme, $create_post
 		return $user_cpt;
 	}
 
+	/**
+	 * Saves a new user variation into the database.
+	 *
+	 * @param array $args {
+	 *     Arguments to add a style variation.
+	 *
+	 *     @type string title Required. Global styles variation name.
+	 *     @type string global_styles Required. Global styles settings as a JSON string.
+	 *     @type string $stylesheet Required. Slug of the theme associated with these global styles.
+	 *     @type string $post_name Optional. Post name.
+	 * }
+	 *
+	 * @return int|WP_Error Post ID of the new variation or error if insertion failed.
+	 */
+	public static function add_user_global_styles_variation( $args ) {
+		$theme = wp_get_theme();
+
+		/*
+		 * Bail early if the theme does not support a theme.json.
+		 *
+		 * Since wp_theme_has_theme_json only supports the active
+		 * theme, the extra condition for whether $theme is the active theme is
+		 * present here.
+		 */
+		if ( $theme->get_stylesheet() === $args['stylesheet'] && ! wp_theme_has_theme_json() ) {
+			return new WP_Error( __( 'Theme does not have theme.json', 'gutenberg' ) );
```

Comment:
Is there any way to make the erorr more helpful? Maybe "Theme does not support theme.json. User style variations only work with themes that support theme.json."